includes/class-wc-gateway-stripe.php

```diff
@@ -1191,4 +1061,204 @@ public function retry_after_error( $response, $order, $retry, $force_save_source
 
         return $this->process_payment( $order->get_id(), true, $force_save_source, $response->error, $previous_error );
 	}
+    /**
+     * Create a new PaymentIntent and attempt to confirm it.
+     *
+     * @param WC_Order $order           The order that is being paid for.
+     * @param object   $prepared_source The source that is used for the payment.
+     * @return object                   An intent (that is either successful or requires an action) or an error.
+     */
+    public function create_and_confirm_intent( $order, $prepared_source ) {
+        // The request for a charge contains metadata for the intent.
+        $full_request = $this->generate_payment_request( $order, $prepared_source );
+
+        $request = array(
+            'source'               => $prepared_source->source,
+            'amount'               => WC_Stripe_Helper::get_stripe_amount( $order->get_total() ),
+            'currency'             => get_woocommerce_currency(),
+            'description'          => $full_request['description'],
+            'metadata'             => $full_request['metadata'],
+            'statement_descriptor' => $full_request['statement_descriptor'],
+            'allowed_source_types' => array(
+                'card',
+            ),
+        );
+
+        if ( $prepared_source->customer ) {
+            $request['customer'] = $prepared_source->customer;
+        }
+
+        // Create an intent that awaits an action.
+        $intent = WC_Stripe_API::request( $request, 'payment_intents' );
+        if ( ! empty( $intent->error ) ) {
+            return $intent;
+		}
+
+		$order->add_order_note( sprintf( __( 'Stripe PaymentIntent initiated (ID: %s)', 'woocommerce-gateway-stripe' ), $intent->id ) );
+
+        // Try to confirm the intent & capture the charge (if 3DS is not required).
+        $request = array(
+			'source' => $request['source'],
+        );
+        $confirmed_intent = WC_Stripe_API::request( $request, "payment_intents/$intent->id/confirm" );
+
+        if ( ! empty( $confirmed_intent->error ) ) {
+			return $confirmed_intent;
+        }
+
+        // Save the intent ID to the order
+		$this->save_intent_to_order( $order, $confirmed_intent );
+
+		// Save a note about the status of the intent.
+		if ( 'succeeded' === $confirmed_intent->status ) {
+			$order->add_order_note( sprintf( __( 'Stripe PaymentIntent succeeded (ID: %s)', 'woocommerce-gateway-stripe' ), $intent->id ) );
```

Comment:
Same comment as above about only adding these to the order notes in a debug mode. Same comment for the below note too.